src/App.vue

```diff
@@ -344,81 +344,97 @@ export default class App extends Mixins(DateMixin, FilingTemplateMixin, LegalApi
+  private async processNameRequest (filing: any): Promise<void> {
+    try {
+      let nameRequest = filing.incorporationApplication.nameRequest
+
+      // ensure we have an NR number
+      if (!nameRequest.nrNumber) {
+        this.nameRequestInvalidType = NameRequestStates.NOT_FOUND
+        this.nameRequestInvalidErrorDialog = true
+        return // go to finally()
+      }
       // fetch NR data
-      const nrResponse = await this.fetchNameRequest(this.nrNumber).catch(error => {
+      const nrResponse = await this.fetchNameRequest(nameRequest.nrNumber).catch(error => {
         console.log('NR error =', error) // eslint-disable-line no-console
-        this.nameRequestInvalidErrorDialog = true
-        throw error // go to catch()
+        // this.nameRequestInvalidErrorDialog = true
+        // throw error // go to catch()
+        return {}
       })
-
       // ensure NR was found
       if (!nrResponse) {
         this.nameRequestInvalidType = NameRequestStates.NOT_FOUND
         this.nameRequestInvalidErrorDialog = true
         return // go to finally()
       }
-
       // ensure NR is valid
       if (!this.isNrValid(nrResponse)) {
         this.nameRequestInvalidType = NameRequestStates.INVALID
         this.nameRequestInvalidErrorDialog = true
         return // go to finally()
       }
-
       // ensure NR is consumable
       const state = this.getNrState(nrResponse)
       if (!state || state !== NameRequestStates.APPROVED) {
         this.nameRequestInvalidType = state || NameRequestStates.INVALID
         this.nameRequestInvalidErrorDialog = true
         return // go to finally()
       }
-
       // if we get this far, the NR is good to go!
       nrResponse.isConsumable = true
-
-      this.setNameRequestState(this.generateNameRequestState(nrResponse, null))
-
-      try {
-        // Retrieve draft filing if it exists for the NR Number specified
-        const draftFiling = await this.fetchDraft()
-
-        // Parse the draft and update NR data into the store if it exists
-        if (draftFiling) {
-          this.parseDraft(draftFiling)
-          this.setNameRequestState(this.generateNameRequestState(nrResponse, +draftFiling.header.filingId))
-        }
-
-        // Initialize Fee Summary
-        this.initEntityFees()
-
-        // Set the resources
-        this.setCertifyStatementResource(CertifyStatementResource.find(x => x.entityType === this.entityType))
-      } catch (error) {
-        console.log('Fetch error =', error) // eslint-disable-line no-console
-        this.fetchErrorDialog = true
-        throw error // go to catch()
-      }
-    } catch (e) {
-      // errors should be handled above
-      // just fall through to finally()
-    } finally {
-      this.haveData = true
-      // wait for things to stabilize, then reset flag
-      Vue.nextTick(() => this.setHaveChanges(false))
+      this.setNameRequestState(this.generateNameRequestState(nrResponse, filing.Id))
+    } catch (ex) {
```

Comment:
Could there be an exception here? Should it be handled? (If not, please add a comment so it doesn't look like missing code.)

Reply:
Also, if lint complains about "unused parameter 'ex'", you can name it "e" (which lint doesn't complain about).

Reply:
should be good now (copied the old comment as it makes sense here)